### PR TITLE
[FIX] account_invoice_fiscal_position_update: filter on invoice lines without display_type

### PR DESCRIPTION
**Pull request by alexis-via**
Fixes PR #465 when the module was migrated to v12

### CODE REVIEW

**Review by legalsylvain — APPROVED** [review]
Makes sense.

thanks.

File: account_invoice_fiscal_position_update/models/account_invoice.py [hunk 1]
```diff
@@ -27,7 +27,8 @@ def fiscal_position_change(self):
         lines_without_product = []
         fp = self.fiscal_position_id
         inv_type = self.type
-        for line in self.invoice_line_ids:
+        for line in self.invoice_line_ids.filtered(
+                lambda l: not l.display_type):
             if line.product_id:
```
Comment (pedrobaeza) [inline account_invoice_fiscal_position_update/models/account_invoice.py:32]: Some people don't  like `filtered` by its performance, but the truth is that you have now 2 comparison at 2 different levels, so let's unify them:
```suggestion
        for line in self.invoice_line_ids:
            if line.product_id or not l.display_type:
```

**Review by pedrobaeza — COMMENTED** [review]
I'm also wondering: does any section or note line contains `product_id`. By default no, so previous code was correct, but your problem is specific to another code that is creating such lines with product filled. Can it be?

**Comment by alexis-via** [comment]
The previous code was not correct because of the "else" and the warning: it would display the warning on section and note lines.

**Comment by pedrobaeza** [comment]
You are totally right. Do you know why Travis is red?

**Comment by pedrobaeza** [comment]
Let's try:

/ocabot merge patch

**Review by chienandalu — APPROVED** [review]

**Review by HaraldPanten — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Fixed problems of the branch in:

08c15179867d90a4ae80e09c85dc9e002fc75388
513553134a7b47c7034bd45a1c30e8d9d9766d36
33afcb8f5ad71332a6f57cecb1fa1ee6848c27e2

/ocabot merge patch

**Pull request merged**